src/highlight-component.js

```diff
@@ -0,0 +1,60 @@
+function HighlightComponent() {
+  this.state = false;
+  this.components = extractComponentsFromPage();
+
+  function extractComponentsFromPage() {
+    return $('[class*="app-c"], [class*="pub-c"], [class*="govuk"]')
+      .toArray()
+      .reduce(function(array, element) {
+        var blockRegex = /(app-c-|pub-c-|govuk-c-|govuk-)([^ _\n]*(?=[ \n]|$))/;
+        var match = $(element).attr('class').match(blockRegex);
+
+        if (match) {
+          array.push({
+            element: element,
+            prefix: match[1],
+            name: match[2]
+          });
+        }
+
+        return array
+      }, []);
+  }
+
+  this.components.forEach(setupComponent.bind(this));
+
+  function setupComponent(component) {
+    var $element = $(component.element);
+    $element.attr('data-component-name', component.name);
+
+    $element.on('click', function() {
+      if (this.state) {
+        window.open( Helpers.documentationUrl(component) );
+      }
+    }.bind(this));
+  }
+
+  chrome.runtime.onMessage.addListener(function (request) {
+    if (request.trigger == 'toggleState') {
+      this.toggleState();
+    }
+  }.bind(this));
+}
+
+HighlightComponent.prototype.toggleState = function () {
+  this.state = !this.state;
+
+  for (var i = 0; i < this.components.length; i++) {
+    $(this.components[i].element).toggleClass('highlight-component', this.state);
```

Comment:
I believe that some components will collapse when we set their position to relative and override their ::after class, but maybe something to follow up later.

Reply:
Yes, we saw this problem with some components that seemed to have a clearfix. We changed the highlight style to use ::before instead and haven't seen any visual problems since. Still, it's something to keep in mind for the future as we'd still be overriding any existing ::before elements with this highlight.